packages/app/src/cli/commands/app/deploy.ts

```diff
@@ -12,6 +12,11 @@ export default class Deploy extends Command {
   static flags = {
     ...cli.globalFlags,
     ...appFlags,
+    'api-key': Flags.string({
+      hidden: false,
+      description: 'The API key of your app.',
+      env: 'SHOPIFY_FLAG_APP_API_KEY',
```

Comment:
I'd make an exception here for the flag name and align with the name that we are using in `.env`, `SHOPIFY_API_KEY`. What do you think?

Reply:
No strong feelings. But note this was copied from `app dev` so we should be consistent for both.

Reply:
So here's an even weirder thing. I'm not even sure where we're using the value from dotenv. Because we seem to fetch the app info in this line: 

https://github.com/Shopify/cli/blob/1f1d9786b6bbab0ed6c56637cd04921be103f823/packages/app/src/cli/services/environment.ts#L132

And that's using cached info in CLI kit store, not the value from .env.

You can prove it by devving with 1 app, then changing the API key in the `.env` file to the key from somewhere else. The value in CLI kit store is used, and the value in `.env` is ignored.

@isaacroldan might be able to provide more context.

Regardless, I think one of the goals of the flag is to allow overriding ENV on a temporary basis. And on those grounds, I think it make sense to leave the flag name with a different ENV variable here, as is done in dev. And if we shorten to `SHOPIFY_API_KEY` in the future, we should update both.

Reply:
`dev` always uses the value from the CLI Kit store.
The value from the `.env` file is used only for `deploy` (we load it [here](https://github.com/Shopify/cli/blob/1f1d9786b6bbab0ed6c56637cd04921be103f823/packages/app/src/cli/services/environment.ts#L226))

Is confusing, but at the time it was the best solution to keep config for `dev` and `deploy` separated. With the presets this will definitely improve